### PR TITLE
[FIX] mail: more friendly pin message dialog

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-3347745

Screenshot 2023-06-06 at 16 57 35
Screenshot 2023-06-06 at 14 32 58

### CODE REVIEW

**Comment by alexkuhn** [comment]
cringe

File: addons/mail/static/src/core_ui/message.js [hunk 1]
```diff
@@ -315,15 +316,24 @@ export class Message extends Component {
...
-        const prompt = this.message.pinned_at
-            ? _t("Are you sure you want to remove this pinned message?")
-            : _t(
-                  "The following message will be pinned to the channel. Are you sure you want to continue?"
-              );
+        const thread = this.message.originThread;
         this.env.services.dialog.add(MessageConfirmDialog, {
+            confirmColor: this.message.pinned_at ? "btn-danger" : undefined,
+            confirmText: this.message.pinned_at
+                ? _t("Yeah, remove it please kthx.")
```
[the remark this answers is not in the document]
Comment (alexkuhn) [reply addons/mail/static/src/core_ui/message.js]: demographic transition done!

File: addons/mail/static/src/core_ui/message.js [hunk 2]
```diff
@@ -315,15 +316,24 @@ export class Message extends Component {
...
+            prompt: this.message.pinned_at
+                ? _t("Enough of that, I guess. You sure you want to remove this pinned message?")
+                : sprintf(
+                      _t(
+                          "Hey, just double checking that you want to pin this message to %(conversation_name)s for posterity and greatness?"
+                      ),
+                      { conversation_name: thread.prefix + thread.displayName }
+                  ),
+            size: "md",
+            title: this.message.pinned_at ? _t("Unpin Message") : _t("Pin It. Pin It Good."),
```
Comment (tde-banana-odoo) [inline addons/mail/static/src/core_ui/message.js:336]: Have to say this reads like bad 18+ movie.

File: addons/mail/static/src/core_ui/message.js [hunk 3]
```diff
@@ -315,15 +316,24 @@ export class Message extends Component {
...
+                ? _t("Yes, remove it please")
+                : _t("Oh yeah. Pin it"),
             message: this.message,
             messageComponent: Message,
-            prompt,
+            prompt: this.message.pinned_at
+                ? _t("Enough of that, I guess. You sure you want to remove this pinned message?")
+                : sprintf(
+                      _t(
+                          "Hey, just double checking that you want to pin this message to %(conversation_name)s for posterity and greatness?"
```
Comment (ticodoo) [inline addons/mail/static/src/core_ui/message.js:331]: I know this is copy pasta text, but maybe if we're allowed to write something else then we could use the slightly less cringe:

```suggestion
                ? _t("Well nothing lasts forever, but are you sure you want to unpin this message?")
                : sprintf(
                      _t(
                          "You sure want this message pinned to %(conversation_name)s forever and ever?"
```

unless we're afraid of upsetting people going through sad periods in life (i.e. death, relationship ending, etc.) :thinking: 

Also both "remove" and "unpin" work in this case, but either way the button should match

File: addons/mail/static/src/core_ui/message.js (hunk 2 above)
Comment (ticodoo) [reply addons/mail/static/src/core_ui/message.js:336]: Doesn't sound sexual in English, but it still sounds awkward... Makes me think of the 80s... https://youtu.be/IIEVqFB4WUo

File: addons/mail/static/src/core_ui/message.js [hunk 4]
```diff
@@ -315,15 +316,22 @@ export class Message extends Component {
...
             message: this.message,
             messageComponent: Message,
-            prompt,
+            prompt: this.message.pinned_at
+                ? _t("Well nothing lasts forever, but are you sure you want to unpin this message?")
+                : sprintf(
+                      _t(
+                          "You sure want this message pinned to %(conversation_name)s forever and ever?"
+                      ),
+                      { conversation_name: thread.prefix + thread.displayName }
```
Comment (oomsveta) [inline addons/mail/static/src/core_ui/message.js:331]: I know you prefer short names 😺
```suggestion
                      _t(
                          "You sure want this message pinned to %(conversation)s forever and ever?"
                      ),
                      { conversation: thread.prefix + thread.displayName }
```

**Review by oomsveta — CHANGES_REQUESTED** [review]
As we are targeting a stable release, please don't forget to re-export the POT files with the latest changes 😇

File: addons/mail/static/src/core_ui/message_confirm_dialog.xml [hunk 5]
```diff
@@ -2,13 +2,13 @@
...
-    <Dialog size="'xl'" title="title">
+    <Dialog size="props.size" title="props.title">
         <p class="mx-3 mb-3" t-esc="props.prompt"/>
         <blockquote class="mx-3 mb-3 fst-normal" style="pointer-events:none;">
             <t t-component="props.messageComponent" message="props.message" hasActions="false"/>
         </blockquote>
         <t t-set-slot="footer">
-            <button class="btn btn-primary me-2" t-on-click="onClickConfirm">Confirm</button>
+            <button class="btn me-2" t-att-class="props.confirmColor" t-on-click="onClickConfirm" t-out="props.confirmText"/>
             <button class="btn btn-secondary me-2" t-on-click="props.close">Cancel</button>
```
Comment (oomsveta) [inline addons/mail/static/src/core_ui/message_confirm_dialog.xml:12]: How does this relate to the task?

Comment (alexkuhn) [reply addons/mail/static/src/core_ui/message_confirm_dialog.xml:12]: Tasks is not just changing text, but also adjusting size of dialog and color of buttons.

**Review by oomsveta — APPROVED** [review]
robodoo r+

**Pull request closed without merging**